src/Microsoft.IdentityModel.Protocols/Configuration/HttpDocumentRetriever.cs

```diff
@@ -64,8 +78,15 @@ public HttpDocumentRetriever(HttpClient httpClient)
             }
             catch (Exception ex)
             {
-                LogHelper.Throw(string.Format(CultureInfo.InvariantCulture, LogMessages.IDX10804, address), typeof(IOException), EventLevel.Error, ex);
-                return null;
+                if (File.Exists(address))
```

Comment:
Why is the HttpDocumentRetriever trying to read from a file? Let alone in an exception block. That doesn't make any sense. If you want to do both in one retriever then rename it and check the address before invoking HttpClient.